src/UrlParams.ts

```diff
@@ -93,14 +95,38 @@ interface UrlParams {
  * @returns The app parameters encoded in the URL
  */
 export const getUrlParams = (
-  query: string = window.location.search,
-  fragment: string = window.location.hash
+  ignoreRoomAlias?: boolean,
+  location: Location = window.location
 ): UrlParams => {
-  const fragmentQueryStart = fragment.indexOf("?");
+  const { href, origin, search, hash } = location;
+
+  let roomAlias: string | undefined;
+  if (!ignoreRoomAlias) {
+    roomAlias = href.substring(origin.length + 1);
+
+    // If we have none, we throw
+    if (!roomAlias) {
+      throw Error("No pathname");
+    }
+    // Delete "/room/" and "?", if present
+    if (roomAlias.startsWith("room/")) {
+      roomAlias = roomAlias.substring("room/".length).split("?")[0];
+    }
+    // Add "#", if not present
+    if (!roomAlias.includes("#")) {
+      roomAlias = `#${roomAlias}`;
+    }
+    // Add server part, if missing
+    if (!roomAlias.includes(":")) {
+      roomAlias = `${roomAlias}:${Config.defaultServerName()}`;
+    }
+  }
```

Comment:
I think we may have miscommunicated, because what I meant by this is that I'd like `getUrlParams` to continue to receive only the query string and fragment as parameters (i.e. only the non-routable parts of the URL), and then let the logic in `App` be the single source of truth for how we parse URL paths and route them in the app.

I believe this would be worthwhile since it would let us implement things like redirects from `/room/:roomAlias` to `/:roomName` more easily by directly applying the tools that React Router gives us, and avoids spreading the routing code across multiple places.

That's the idea at least, though I recognize that the room ID sometimes taking the form of `#?roomId=...` does complicate this… Do you have any thoughts on this?

Reply:
I see, another problem might be that I am not really sure how the redirecting would behave in embedded mode - I am scared you might get unnecessary loading screens then which is why I put everything in here to avoid the redirects

To be frank I am not 100% sure I see the problem in what you're describing - I might need to think through more closely, but I see the paragraph above as a bigger issue